core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1032,7 +1032,9 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
   /**
    * Like fork but handles an error with the provided handler.
    */
-  final def forkWithErrorHandler(handler: E => UIO[Any])(implicit trace: ZTraceElement): URIO[R, Fiber.Runtime[E, A]] =
+  final def forkWithErrorHandler[R1 <: R](handler: E => URIO[R1, Unit])(implicit
```

Comment:
```suggestion
  final def forkWithErrorHandler[R1 <: R](handler: E => URIO[R1, Any])(implicit
```

Reply:
The more general type signature here will allow people to use an error handler that returns an arbitrary value without having to call `unit`.